r/README.Rmd

```diff
@@ -16,48 +18,77 @@ knitr::opts_chunk$set(
+
+The `arrow` package exposes an interface to the Arrow C++ library.
 
 ## Installation
 
-You first need to install the C++ library:
+Installing `arrow` is a two-step process: first install the Arrow C++ library, then the R package.
 
-### macOS
+### Release version
 
-On macOS, you may install using homebrew:
+The current release of Apache Arrow is 0.13.
 
-```
+On macOS, you may install the C++ library using [Homebrew](https://brew.sh/):
+
+```shell
 brew install apache-arrow
 ```
 
-### From source
+On Linux, see the [Arrow project installation page](http://arrow.apache.org/install/).
+
+Then, you can install the release version of the package from GitHub using the [`remotes`](https://remotes.r-lib.org/) package. From within an R session,
+
+```r
+# install.packages("remotes") # Or install "devtools", which includes remotes
+remotes::install_github("apache/arrow/r", ref="76e1bc5dfb9d08e31eddd5cbcc0b1bab934da2c7")
+```
 
-First install a release build of the C++ bindings to arrow.
+or if you prefer to stay at the command line,
 
 ```shell
-git clone https://github.com/apache/arrow.git
-cd arrow/cpp && mkdir release && cd release
+R -e 'remotes::install_github("apache/arrow/r", ref="76e1bc5dfb9d08e31eddd5cbcc0b1bab934da2c7")'
+```
+
+### Development version
+
+First, clone the repository and install a release build of the C++ library.
 
-# It is important to statically link to boost libraries
-cmake .. -DARROW_PARQUET=ON -DCMAKE_BUILD_TYPE=Release -DARROW_BOOST_USE_SHARED:BOOL=Off
+```shell
+git clone https://github.com/apache/arrow.git
+mkdir arrow/cpp/build && cd arrow/cpp/build
+cmake .. -DARROW_PARQUET=ON -DARROW_BOOST_USE_SHARED:BOOL=Off
 make install
 ```
 
-## Then the R package
+Then, you can install the R package and its dependencies from the git checkout:
+
+```shell
+cd ../../r
+R -e 'install.packages("remotes"); remotes::install_deps()'
+R CMD INSTALL .
+```
 
-### From github (defaults to master branch)
+If the package fails to install/load with an error like this:
 
-```r
-library("devtools")
-devtools::install_github("apache/arrow/r")
 ```
+** testing if installed package can be loaded from temporary location
+Error: package or namespace load failed for 'arrow' in dyn.load(file, DLLpath = DLLpath, ...):
+unable to load shared object '/Users/you/R/00LOCK-r/00new/arrow/libs/arrow.so':
+dlopen(/Users/you/R/00LOCK-r/00new/arrow/libs/arrow.so, 6): Library not loaded: @rpath/libarrow.14.dylib
+```
+
+try setting the environment variable `LD_LIBRARY_PATH` to wherever Arrow C++ was put in `make install`, e.g. `export LD_LIBRARY_PATH=/usr/local/lib`, and retry installing the R package. An alternative solution on macOS is to rebuild the C++ library adding the `-DARROW_INSTALL_NAME_RPATH=OFF` option to the `cmake` command.
+
+For any other build/configuration challenges, see the [C++ developer guide](http://arrow.apache.org/docs/developers/cpp.html#building).
```

Comment:
Could you use `https`?

Reply:
Interesting, I'm getting these `http` URLs because I'm copying from the web browser, and it's not redirecting to `https` like most sites do. I made https://issues.apache.org/jira/browse/ARROW-5417 for that.